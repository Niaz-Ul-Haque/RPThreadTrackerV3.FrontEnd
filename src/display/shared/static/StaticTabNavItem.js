import React from 'react';
import PropTypes from 'prop-types';
import {
	NavItem, NavLink
} from 'reactstrap';

const propTypes = {
	tabId: PropTypes.string.isRequired,
	activeTab: PropTypes.string.isRequired,
	setActiveTab: PropTypes.func.isRequired,
	iconId: PropTypes.string.isRequired,
	title: PropTypes.string.isRequired
};

class StaticTabNavItem extends React.Component {
	constructor() {
		super();
		this.onClick = this.onClick.bind(this);
	}
	onClick(e, tabId) {
		e.preventDefault();
		this.props.setActiveTab(tabId);
	}
	render() {
		const {
			tabId,
			activeTab,
			iconId,
			title
		} = this.props;
		return (
			<NavItem>
				<NavLink
					href={`#${tabId}`}
					className={activeTab === tabId ? 'active' : ''}
					onClick={(e) => { this.onClick(e, tabId); }}
				>
					<span className="d-none d-xl-inline"><i className={iconId ? `fas fa-${iconId}` : ''} /></span> {title}
				</NavLink>
			</NavItem>
		);
	}
}
StaticTabNavItem.propTypes = propTypes;

export default StaticTabNavItem;
