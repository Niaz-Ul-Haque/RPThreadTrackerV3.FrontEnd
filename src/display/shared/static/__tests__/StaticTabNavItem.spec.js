// #region imports
import React from 'react';
import { shallow } from 'enzyme';
import StaticTabNavItem from '../StaticTabNavItem';
// #endregion imports

const createTestProps = propOverrides => ({
	href: '/test/mock-tab',
	title: 'Mock Tab',
	...propOverrides
});
describe('rendering', () => {
	describe('snapshots', () => {
		it('should render valid snapshot', () => {
			const props = createTestProps();
			const jsx = (<StaticTabNavItem {...props} />);
			const element = shallow(jsx);
			expect(element).toMatchSnapshot();
		});
	});
});
