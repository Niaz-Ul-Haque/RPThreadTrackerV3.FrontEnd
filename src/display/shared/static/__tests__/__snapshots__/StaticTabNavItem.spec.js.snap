// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering snapshots should render valid snapshot 1`] = `
<NavItem
  tag="li"
>
  <NavLink
    activeClassName="active"
    aria-current="page"
    data-spec="static-tab-nav-item-navlink"
    to="/test/mock-tab"
  >
    Mock Tab
  </NavLink>
</NavItem>
`;
